twitter-clone/src/AuthContext.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, useState, useContext } from 'react';
+
+export interface User {
+  username: string;
+  id: string;
+  name: string;
+  email: string;
+}
+
+interface AuthContextType {
+  user: User | null;
+  logIn: (userData: User) => void;
+  logOut: () => void;
+}
+
+const defaultAuthContext = {
+  user: null,
+  logIn: () => {},
+  logOut: () => {},
+};
+
+export const useAuth = (): AuthContextType => {
```

Comment:
better name it useAuthContext, cos for now it seems to be some kind of hook for authorization.